src/ert/storage/local_ensemble.py

```diff
@@ -715,6 +910,59 @@ def load_all_gen_kw_data(
 
         return dataframe.sort_index(axis=1)
 
+    def _validate_parameters_dataset(self, group: str, dataset: xr.Dataset) -> None:
+        if "values" not in dataset.variables:
+            raise ValueError(
+                f"Dataset for parameter group '{group}' "
+                f"must contain a 'values' variable"
+            )
+
+        if dataset["values"].size == 0:
+            raise ValueError(
+                f"Parameters {group} are empty. Cannot proceed with saving to storage."
+            )
+
+        if dataset["values"].ndim >= 2 and dataset["values"].values.dtype == "float64":
+            logger.warning(
+                "Dataset uses 'float64' for fields/surfaces. Use 'float32' to save memory."
+            )
+
+        if group not in self.experiment.parameter_configuration:
+            raise ValueError(f"{group} is not registered to the experiment.")
+
+    def _assert_dataset_not_already_created(
+        self, group: str, realization: Optional[int] = None
+    ) -> None:
+        path_in_base_folder = self._path / f"{group}.nc"
+        if os.path.exists(path_in_base_folder):
+            f"There already exists a combined dataset for parameter group {group}"
+            f" for group {group} @ {path_in_base_folder}. Parameters should be saved only once."
+
+        if realization is not None:
+            path_in_real_folder = self._realization_dir(realization) / f"{group}.nc"
+            if os.path.exists(path_in_real_folder):
+                raise KeyError(
+                    "Detected attempt at overwriting already saved parameter"
+                    f" for group {group} @ {path_in_real_folder}. Parameters should be saved only once."
+                )
+
+    def save_parameters_combined(self, group: str, dataset: xr.Dataset) -> None:
```

Comment:
Only called once, as per other comment, might be removed? Otherwise, it will `@require_write`